### PR TITLE
Builder API integration tests

### PR DESCRIPTION
**Pull request by baumanj**
Test the builder API via the builder dev studio both locally and in travis

Adapted from https://github.com/habitat-sh/habitat/pull/4223

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by raskchanky** [comment]
If you use `git subtree split` and `git subtree add`, you can bring the branch over from the previous repo while preserving the git history.

File: components/builder-http-gateway/src/http/middleware.rs [hunk 1]
```diff
@@ -170,6 +171,7 @@ impl Key for Authenticated {
 
 impl BeforeMiddleware for Authenticated {
     fn before(&self, req: &mut Request) -> IronResult<()> {
+        aobafbp
```
Comment (raskchanky) [inline components/builder-http-gateway/src/http/middleware.rs]: What's this?

File: components/builder-http-gateway/src/http/middleware.rs [hunk 2]
```diff
@@ -204,7 +206,11 @@ impl BeforeMiddleware for Authenticated {
                         // Check to see if this is a valid github token, and create (or
                         // update) a session. This is a temporary fix until we can roll out
                         // and migrate clients to our own personal access tokens.
-                        session_create_github(req, token)?
+                        if env::var_os("HAB_FUNC_TEST").is_some() {
```
Comment (raskchanky) [inline components/builder-http-gateway/src/http/middleware.rs]: In my original PR, I had a lengthy comment in this file explaining the nature of our session tokens and why our current implementation was broken but it seems to have disappeared.  Any idea why or where it went to?

**Comment by raskchanky** [comment]
Closing in favor of https://github.com/habitat-sh/builder/pull/124 which preserves the full git history.

File: components/builder-http-gateway/src/http/middleware.rs (hunk 1 above)
Comment (baumanj) [reply components/builder-http-gateway/src/http/middleware.rs]: Unintentionally committed. I was in the middle of testing https://github.com/habitat-sh/builder/pull/109 and intentionally causing build failures

**Pull request closed without merging**